xapian-core/cluster/cluster.cc

```diff
@@ -132,92 +141,357 @@ TermListGroup::get_termfreq(const string &tname) {
 }
 
 doccount
-ClusterSet::num_of_clusters() {
-    LOGCALL(API, doccount, "ClusterSet::num_of_clusters()", "");
+ClusterSet::size() {
```

Comment:
This method (and many others) should be `const` - they don't touch the internal state of the object, and so should be usable on a `const` object - e.g. `cluster_set.size()` should be callable in a function which takes a `const ClusterSet& cluster_set` parameter.